src/libstd/io/mod.rs

```diff
@@ -287,15 +525,31 @@ pub trait Read {
 
 /// A trait for objects which are byte-oriented sinks.
 ///
-/// The `write` method will attempt to write some data into the object,
-/// returning how many bytes were successfully written.
+/// Implementors of the `Write` trait are sometimes called 'writers'.
+///
+/// Readers are defined by two required methods, `write()` and `flush()`:
 ///
-/// The `flush` method is useful for adaptors and explicit buffers themselves
-/// for ensuring that all buffered data has been pushed out to the "true sink".
+/// * The `write()` method will attempt to write some data into the object,
+///   returning how many bytes were successfully written.
 ///
-/// Writers are intended to be composable with one another. Many objects
-/// throughout the I/O and related libraries take and provide types which
-/// implement the `Write` trait.
+/// * The `flush()` method is useful for adaptors and explicit buffers
+///   themselves for ensuring that all buffered data has been pushed out to the
+///   'true sink'.
+///
+/// Writers are intended to be composable with one another. Many implementors
+/// throughout `std::io` take and provide types which implement the `Write`
+/// trait.
+///
+/// # Examples
+///
+/// ```
+/// use std::io::Write;
+///
+/// let mut buffer = Vec::new();
```

Comment:
This may want to have an explicit comment explaining that in-memory buffers (`Vec`, slices, etc), all implement the `Write` trait (e.g. it's not always pure I/O)